src/vm/dllimport.cpp

```diff
@@ -6096,7 +6104,42 @@ static void DetermineLibNameVariations(const char* const** libNameVariations, in
         *numberOfVariations = COUNTOF(NameOnly);
     }
 }
-#endif
+#else // FEATURE_PAL
+static void DetermineLibNameVariations(const char* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
+{
+    bool endWithDot = !!libName.EndsWith(W("."));
```

Comment:
Now that I am looking at this again ... caching these in bool variables makes this less efficient than it needs to be. If you put these directly into the if statement without caching them in locals like this:

```
if (!libName.Find(libName.Begin(), W('.')) ||
    libName.EndsWith(W(".")) ||
    libName.EndsWithCaseInsensitive(W(".dll")) || libName.EndsWithCaseInsensitive(W(".exe"))
{
```

It will be more efficient because of it will stop at `!libName.Find(libName.Begin(), W('.'))` for names without dot; and the remaining two statements won't be executed at all.

It also avoids the `!!` wart.